transport/src/main/java/io/netty5/channel/ChannelOutboundBuffer.java

```diff
@@ -200,6 +200,9 @@ private void decrementPendingOutboundBytes(long size, boolean invokeLater, boole
     }
 
     private static long total(Object msg) {
+        if (msg instanceof Buffer) {
+            return ((Buffer) msg).readableBytes();
+        }
         if (msg instanceof ByteBufConvertible) {
             return ((ByteBufConvertible) msg).asByteBuf().readableBytes();
```

Comment:
maybe I missed something, I'm not sure I understand.
Indeed, the thing which is expensive is the call to _asByteBuf():_

```
        if (msg instanceof ByteBufConvertible) {
            return ((ByteBufConvertible) msg).asByteBuf().readableBytes();
        }
```
And if I'm correct, some instances of a _Buffer_, like a _io.netty5.buffer.api.bytebuffer.NioBuffer_, are also an instance of a ByteBufConvertiable.

So, if we do the "_if (msg instanceof Buffer)_" test lastly, like:

```
        if (msg instanceof ByteBufConvertible) {
            return ((ByteBufConvertible) msg).asByteBuf().readableBytes();
        }
        if (msg instanceof FileRegion) {
            return ((FileRegion) msg).count();
        }
        if (msg instanceof ByteBufHolder) {
            return ((ByteBufHolder) msg).content().readableBytes();
        }
         if (msg instanceof Buffer) {
            return ((Buffer) msg).readableBytes();
        }
```

then, when the msg parameter is a Buffer instance like "_io.netty5.buffer.api.bytebuffer.NioBuffer_", then the first  "_if msg instanceof ByteBufConvertible)_" test would always win, and in this case the optimization would never take place.
Am I correct ?

Reply:
right, I'm suggesting to move the most expensive block to last position.

Reply:
ah ok, understood. I'll update the PR. thanks for the review !

Reply:
The `asByteBuf` method is only expensive on the `Buffer` implementations. Calling `asByteBuf` on `ByteBuf` just returns itself. So the important thing is just that the `instanceof Buffer` check comes first. No need to change it back, though, as the order is otherwise harmless.